nifi-extension-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/NettyEventServerFactory.java

```diff
@@ -202,14 +200,15 @@ private void setChannelOptions(final AbstractBootstrap<?, ?> bootstrap) {
     }
 
     private AbstractBootstrap<?, ?> getBootstrap() {
+        NettyTransports.NettyTransport nettyTransport = this.getNettyTransport();
```

Comment:
Instead of interacting with the `NettyTransport` record, it seems like this could be changed to have a `getDatagramChannelClass()` and `getServerSocketClass()` method that avoids exposing the record class itself.

Reply:
Agreed. I'll remove the getNettyTransport() method.